### PR TITLE
chore: Add automatic deployment to Github pages

### LINKED ISSUE #1218: Deploy front-end to gh-pages
The badgeyay project deploys the ember frontend to gh-pages. Can we do the same here, e.g. using this https://www.npmjs.com/package/ember-cli-deploy-github-pages ?

### PR DESCRIPTION
**Pull request by mayank8318**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Adds automatic deployment to GitHub pages

#### Changes proposed in this pull request:
- Added a script to automatically deploy the repo to GitHub pages
- Executing this script in the `after-script` section of Travis

Fixes #1218

### CODE REVIEW

**Comment by mayank8318** [comment]
@bhaveshAn @srv-twry @Kreijstal @pradeepgangwar @niranjan94 @SaptakS Please review :)

File: config/environment.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ module.exports = function(environment) {
     appName                  : process.env.APP_NAME || 'Open Event',
     modulePrefix             : 'open-event-frontend',
     environment,
-    rootURL                  : '/',
+    rootURL                  : 'open-event-frontend',
```
Comment (niranjan94) [inline config/environment.js]: This should not be hardcoded in the repo

File: .travis.yml [hunk 2]
```diff
@@ -25,7 +25,9 @@ addons:
 
 before_install: yarn global add greenkeeper-lockfile@1
 before_script: greenkeeper-lockfile-update
-after_script: greenkeeper-lockfile-upload
+after_script:
+  - greenkeeper-lockfile-upload
+  - bash ./gh_pages.sh
```
Comment (niranjan94) [inline .travis.yml]: Use https://docs.travis-ci.com/user/deployment/script/ instead. Travis allows you to specify the branch on which it should run too.

File: gh_pages.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+ember install ember-cli-github-pages
+git checkout $TARGET_BRANCH || git checkout --orphan $TARGET_BRANCH
+rm -rf `bash -c "ls -a | grep -vE '\.gitignore|\.git|node_modules|bower_components|(^[.]{1,2}/?$)'"` && touch .gitkeep
+git add .
+git commit -m "Deploy to GitHub Pages: ${SHA}"
+git checkout $SOURCE_BRANCH
+
+# Build the ember app to Target Branch
+ember $TARGET_BRANCH:commit --message "Deploy to GitHub Pages: ${SHA}"
```
Comment (niranjan94) [inline gh_pages.sh]: Make sure there exists only _one_ commit on the `gh-pages` repo. Every commit should just overwrite it. And you should force push to it.

Comment (niranjan94) [reply gh_pages.sh]: This handled nicely by https://docs.travis-ci.com/user/deployment/pages/. I don't see why you'd need to use `ember-cli-github-pages` for this.

Comment (niranjan94) [reply gh_pages.sh]: This handled nicely by https://docs.travis-ci.com/user/deployment/pages/. I don't see why you'd need to use `ember-cli-github-pages` for this. 

You can just set the correct path at build time, build it and just point travis to deploy the distribution folder to gh-pages

File: config/environment.js (hunk 1 above)
Comment (mayank8318) [reply config/environment.js]: But the addon needs it to be this way thoughhttps://stackoverflow.com/a/24788051/6870128

Comment (niranjan94) [reply config/environment.js]: Two things.

Comment (niranjan94) [reply config/environment.js]: 1. You don't really need an addon for such a simple task. Just ember's build command and travis's github pages release feature is more than enough.
2. You can modify that just for the build. Not hard coded in the repo. (Hint. you can use an env var to set it)

**Comment by mayank8318** [comment]
@niranjan94 I have added the Travis direct deployment method but I don't have the GitHub token. If this works I will delete the script. Please review :)

File: .travis.yml [hunk 4]
```diff
@@ -37,3 +37,11 @@ after_success:
   - bash <(curl -s https://codecov.io/bash)
  # - bash scripts/pr_deploy.sh
  # - bash kubernetes/travis/deploy.sh
+
+deploy:
```
Comment (niranjan94) [inline .travis.yml]: Specify which directory to deploy to gh-pages.

File: .travis.yml [hunk 5]
```diff
@@ -37,3 +37,11 @@ after_success:
   - bash <(curl -s https://codecov.io/bash)
  # - bash scripts/pr_deploy.sh
  # - bash kubernetes/travis/deploy.sh
```
Comment (niranjan94) [inline .travis.yml]: Where are you setting the rootPath and building ?

**Comment by niranjan94** [comment]
@mayank8318 you can remove the script already and make the changes requested above. I'll set the github token on travis. I'll review it and we'll merge and test the full flow.

**Comment by mayank8318** [comment]
@niranjan94 I only need to specify the folder as `dist` right?

File: .travis.yml (hunk 4 above)
Comment (mayank8318) [reply .travis.yml]: Added

**Comment by niranjan94** [comment]
And you need to set the root path and run the ember build command. Both are missing :)

**Comment by mayank8318** [comment]
@niranjan94 Added the rootUrl as an env variable(included it in the .env.example)

File: .env.example [hunk 6]
```diff
@@ -1,2 +1,3 @@
 GOOGLE_API_KEY="Sample Key"
 API_HOST=http://127.0.0.1:5000
+rootURL='open-event-frontend'
```
Comment (niranjan94) [inline .env.example]: No user would need to set rootUrl for local testing. So, remove it from here. it is only required for gh-pages deployment. So, just set it on travis. (See next comment on how)

File: .travis.yml [hunk 7]
```diff
@@ -32,8 +32,18 @@ script:
   - node scripts/l10n.js generate
   - yarn lint:scss
   - COVERAGE=true yarn test
+  - ember build
```
Comment (niranjan94) [inline .travis.yml]: Build the production version please.

```bash
ROOT_URL=open-event-frontend ember build -prod
```

File: config/environment.js [hunk 8]
```diff
@@ -13,7 +13,7 @@ module.exports = function(environment) {
     appName                  : process.env.APP_NAME || 'Open Event',
     modulePrefix             : 'open-event-frontend',
     environment,
-    rootURL                  : '/',
+    rootURL                  : process.env.rootURL || '/',
```
Comment (niranjan94) [inline config/environment.js]: Follow the same naming scheme for environment variables. `ROOT_URL`

**Pull request merged**